projects/core/src/cart/store/actions/cart.action.ts

```diff
@@ -71,6 +71,11 @@ export class MergeCartSuccess implements Action {
   constructor(public payload: any) {}
 }
 
+export class ResetCartDetails implements Action {
```

Comment:
we don't need this action. In `reducer/index.ts`, we have this:
```
export function clearCartState(
  reducer: ActionReducer<any>
): ActionReducer<any> {
  return function(state, action) {
    if (action.type === LOGOUT || action.type === PLACE_ORDER_SUCCESS) {
      state = undefined;
    }
    return reducer(state, action);
  };
}
```
It does the same thing. We only need to add `LANGUAGE_CHANGE`/`CURRENCY_CHANGE` into the action type.

Reply:
ResetCartDetails is more sophisticated than just clearing the whole cart state. It leaves the guid and code (see cart.reducer), so anonymous user doesn't loose his cart after langauge change.

This sophisticated modification of the state is performed in the cart reducer, and the actions `LANGUAGE_CHANGE/CURRENCY_CHANGE` dispatch explicitly the `ResetCartDetails` action in the cart.effect, so it's debuggable in the Redux dev tools.